lib/semian/lru_hash.rb

```diff
@@ -83,16 +86,30 @@ def clear_unused_resources
     return unless @lock.try_lock
     # Clears resources that have not been used in the last 5 minutes.
     begin
+      timer_start = Process.clock_gettime(Process::CLOCK_MONOTONIC)
+
+      payload = {
+          size: @table.size,
+          examined: 0,
+          cleared: 0,
+          elapsed: nil,
+      }
+
+      stop_time = Time.now - MINIMUM_TIME_IN_LRU # Don't process resources updated after this time
       @table.each do |_, resource|
-        break if resource.updated_at + MINIMUM_TIME_IN_LRU > Time.now
+        payload[:examined] += 1
+        break if resource.updated_at > stop_time
```

Comment:
I know you're just moving this code around, but why is this a `break` rather than a `next`?

Reply:
This is an LRU cache, so the table should be ordered by update time.
I was going to say that if we see a resource that's newer than the stop time, we should assume that all elements after that are also newer, but I'm not 100% convinced of that, since `get` doesn't update the `updated_at` field.
I'm going to look into this.

Reply:
Thanks for catching that @csfrancis.
Indeed, we should be updating `updated_at` in the call to `get` when we re-order the LRU.
I've made the fix and added a test for monotonicity.

Reply:
Nice catch!